fcrepo-exts/fcrepo-doctor
# Add object analysis for invalid binary description subjects

# What does this Pull Request do?

- Builds out a framework for scanning objects
- Adds a scanner for identifying binary descriptions with invalid subjects

# How should this be tested?

1. Create binaries in Fedora 6 and add triples to their descriptions
2. Verify in the files on disk that you see subjects that end with `fcr:metadata`
3. Execute this project on the OCFL root: `java -jar target/fcrepo-doctor-0.0.1-SNAPSHOT.jar -r <PATH-TO-FEDORA-OCFL-ROOT> -t /var/tmp`
4. Examine the output file (`fcrepo-doctor-problems.json` in the current directory) to confirm that it identified all of the expected resources

# Additional Notes:

This PR does **not** implement the remediation portion yet, and simply produces a file that identifies all affected objects.

# Interested parties

@fcrepo/committers
